newIDE/app/src/Utils/GDevelopServices/Play.js

```diff
@@ -0,0 +1,239 @@
+// @flow
+import axios from 'axios';
+import { GDevelopPlayApi } from './ApiConfigs';
+
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+
+export type Leaderboard = {|
+  id: string,
+  gameId: string,
+  name: string,
+  sort: 'DESC' | 'ASC',
+  startDatetime: string,
+  deletedAt?: string,
+  playerUnicityDisplayChoice: 'PREFER_UNIQUE' | 'PREFER_NON_UNIQUE' | 'FREE',
+|};
+
+export type LeaderboardEntry = {|
+  id: string,
+  leaderboardId: string,
+  playerId?: string,
+  playerName: string,
+  createdAt: string,
+  score: number,
+  deletedAt?: string,
+  outdatedAt?: string,
+|};
+
+export type LeaderboardDisplayData = {|
+  +id: string,
+  +playerName: string,
+  +createdAt: string,
+  +score: number,
+|};
+
+export type LeaderboardSortOption = 'ASC' | 'DESC';
+export type LeaderboardPlayerUnicityDisplayOption =
+  | 'FREE'
+  | 'PREFER_UNIQUE'
+  | 'PREFER_NON_UNIQUE';
+
+export type LeaderboardExtremePlayerScore = {|
+  leaderboardId: string,
+  playerId?: string,
+  playerName: string,
+  relatedEntryCreatedAt: string,
+  score: number,
+  relatedEntryId: string,
+|};
+
+export const extractEntryDisplayData = ({
+  playerName,
+  id,
+  score,
+  createdAt,
+}: LeaderboardEntry): LeaderboardDisplayData => ({
+  id,
+  createdAt,
+  playerName,
+  score,
+});
+
+export const extractExtremeScoreDisplayData = ({
+  playerName,
+  relatedEntryId,
+  score,
+  relatedEntryCreatedAt,
+}: LeaderboardExtremePlayerScore): LeaderboardDisplayData => ({
+  id: relatedEntryId,
+  createdAt: relatedEntryCreatedAt,
+  playerName,
+  score,
+});
+
+export const listGameLeaderboards = (gameId: string): Promise<Leaderboard[]> =>
+  axios
+    .get(`${GDevelopPlayApi.baseUrl}/game/${gameId}/leaderboards`)
+    .then(response => response.data);
+
+export const extractNextPageUriFromLinkHeader = (
+  linkHeader: string
+): ?string => {
+  const links = linkHeader.split(',').map(link => link.trim());
+  const mapRelationToUri = links.reduce((acc, link) => {
+    const relationRegexMatch = link.match(/;\srel="(\w*)"/);
+    const uriMatch = link.match(/^<(.*)>/);
+    if (acc && relationRegexMatch && uriMatch) {
+      acc[relationRegexMatch[1]] = uriMatch[1];
+    }
+    return acc;
+  }, {});
+  if (Object.keys(mapRelationToUri).includes('next')) {
+    return mapRelationToUri.next;
+  }
+  return null;
+};
+
+export const listLeaderboardEntries = async (
+  gameId: string,
+  leaderboardId: string,
+  options: {| pageSize: number, onlyBestEntry: boolean, forceUri: ?string |}
+): Promise<{|
+  entries: LeaderboardEntry[] | LeaderboardExtremePlayerScore[],
+  nextPageUri: ?string,
+|}> => {
+  const uri =
+    options.forceUri || `/game/${gameId}/leaderboard/${leaderboardId}/entries`;
+  // $FlowFixMe
```

Comment:
what's that for?

Reply:
This is because of the fixed-in-0.159-version-of-flow bug: https://github.com/facebook/flow/issues/5294.

The fact that I use the response below in the `extractNextPageUriFromLinkHeader` method make it fail